tests/common/test_utils.py

```diff
@@ -408,6 +419,37 @@ def construct_run_config_result(avg_gpu_metric_values,
 
     return run_config_result
 
+class ConfigCommandProfileSubClass(ConfigCommandProfile):
```

Comment:
These optimizations are nice, but I don't think they are what should have been optimized.
I took a closer look at just one test -- test_sorted_results.py:test_add_result().  That one test creates 210 constraint managers(!). I don't see any reason for any test to ever create more than one. Once you fix that, the latency of the full CLI objects should be totally fine without the need for this optimization.

All of the construct_* functions should likely take an optional (or required?) constraint_manager so that each level doesn't have to create a new one.

Reply:
Note that the 210 calls was from your old changes, not your new ones. I see now that you have magic mocked out most of the create calls in the new code.  I had assumed based on our discussion yesterday that you had only optimized the CLI code.

I don't see any reason to fake out the CLI code. As long as no test creates more than one constraint_manager I can't imagine that the full CLI code takes more than 1 second of total test run time.